### PR TITLE
eliminated some ASTImporterTest warnings

### PR DESCRIPTION
**Pull request by balazske**

### CODE REVIEW

**Comment by balazske** [comment]
One test fails that fails on ctu-clang6 too:
```
********************
FAIL: Clang :: Misc/warning-flags.c (5250 of 11957)
******************** TEST 'Clang :: Misc/warning-flags.c' FAILED ********************
Script:
--
diagtool list-warnings > /home/ebalkri/clang/clang6/build/tools/clang/test/Misc/Output/warning-flags.c.tmp 2>&1
/home/ebalkri/clang/clang6/build/bin/FileCheck --input-file=/home/ebalkri/clang/clang6/build/tools/clang/test/Misc/Output/warning-flags.c.tmp /home/ebalkri/clang/clang6/llvm/tools/clang/test/Misc/warning-flags.c
--
Exit Code: 1

Command Output (stderr):
--
/home/ebalkri/clang/clang6/llvm/tools/clang/test/Misc/warning-flags.c:21:8: error: expected string not found in input
CHECK: Warnings without flags (79):
       ^
/home/ebalkri/clang/clang6/build/tools/clang/test/Misc/Output/warning-flags.c.tmp:1:1: note: scanning from here
Warnings with flags (1311):
^
```

File: unittests/AST/ASTImporterTest.cpp [hunk 1]
```diff
@@ -923,11 +945,11 @@ TEST(ImportDecl, ImportUsingDecl) {
                          usingDecl())))))));
 }
 
-TEST(ImportDecl, ImportRecordDeclInFuncParams) {
+TEST(ImportDecl, DISABLED_ImportRecordDeclInFuncParams) {
```
Comment (balazske) [inline unittests/AST/ASTImporterTest.cpp]: I found that this test is only applicable to C code not C++.
When run with C code another type of error is generated (this needs to be corrected later).

**Comment by Xazax-hun** [comment]
Alternatively, we could suppres unused variable warnings like:
```
int x;
(void)x;
```
I am not sure which option is better though.

**Comment by balazske** [comment]
I do not want to modify the test code, `(void)x` is not the same expression as `x`.

**Review by martong — APPROVED** [review]
LGTM.

**Review by Xazax-hun — APPROVED** [review]

**Comment by balazske** [comment]
Phabricator link:
https://reviews.llvm.org/D47459

**Comment by balazske** [comment]
Backport:
https://github.com/Ericsson/clang/pull/412

**Pull request merged**